lib/mongo/retryable/base_worker.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+# Copyright (C) 2015-2020 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  module Retryable
+
+    # The abstract superclass for workers employed by Mongo::Retryable.
+    #
+    # @api private
+    #
+    # @since 2.19.0
+    class BaseWorker
+      extend Forwardable
+
+      def_delegators :retryable,
+        :client,
+        :cluster,
+        :select_server
+
+      # @return [ Mongo::Retryable ] retryable A reference to the client object
+      #   that instatiated this worker.
+      attr_reader :retryable
+
+      # Constructs a new worker.
+      #
+      # @example Instantiating a new read worker
+      #   worker = Mongo::Retryable::ReadWorker.new(self)
+      #
+      # @example Instantiating a new write worker
+      #   worker = Mongo::Retryable::WriteWorker.new(self)
+      #
+      # @param [ Mongo::Retryable ] retryable The client object that is using
+      #   this worker to perform a retryable operation
+      def initialize(retryable)
+        @retryable = retryable
+      end
+
+      private
+
+      # Returns a list of exception classes that are generally retryable. 
```

Comment:
```suggestion
      # @return [ Array<Mongo::Error> ] Array of exception classes that are generally retryable. 
```
I know it contradicts the Rails style, but I believe it is beneficial to add type information to the documenting comments - `@param` and `@return` in particular. While it is not required for the private methods, it is still helpfult for tools like [solargraph](https://solargraph.org/guides/yard)